spec/support/mock_auth_api_helper.rb

```diff
@@ -0,0 +1,54 @@
+module MockAuthApiHelper
+  MockAuthApiNotSetup = Class.new(Exception)
+
+  def mock_auth_api_client(setup)
+    @old_service = ServiceRegistry.service(:auth_api_client)
```

Comment:
So I've verified that the initialisers are not run before every test. I'll leave this manual tear down here for the moment.